src/state.rs

```diff
@@ -131,6 +138,8 @@ impl ProcessState for DefaultProcessState {
             wasi_stderr: None,
             initialized: false,
             registry: self.registry.clone(),
+            plugins: self.plugins.clone(),
+            plugin_state: HashMap::new(),
```

Comment:
Should we clone the plugin state here? Or put the whole plugin state in an Arc?
The only problem is that we cannot `downcast_mut` when the state is in an Arc, or even borrow it mutably.